push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/apns/PushyApnsSender.java

```diff
@@ -208,35 +212,42 @@ private ApnsClient buildApnsClient(final iOSVariant iOSVariant) {
         // this check should not be needed, but you never know:
         if (iOSVariant.getCertificate() != null && iOSVariant.getPassphrase() != null) {
 
-            // add the certificate:
-            try {
-                final ByteArrayInputStream stream = new ByteArrayInputStream(iOSVariant.getCertificate());
-
-                final ApnsClientBuilder builder = new ApnsClientBuilder();
-                builder.setClientCredentials(stream, iOSVariant.getPassphrase());
-
-                if (ProxyConfiguration.hasHttpProxyConfig()) {
-                    if (ProxyConfiguration.hasBasicAuth()) {
-                        String user =  ProxyConfiguration.getProxyUser();
-                        String pass = ProxyConfiguration.getProxyPass();
-                        builder.setProxyHandlerFactory(new HttpProxyHandlerFactory(ProxyConfiguration.proxyAddress(), user, pass));
-                    } else {
-                        builder.setProxyHandlerFactory(new HttpProxyHandlerFactory(ProxyConfiguration.proxyAddress()));
-                    }
+            // only process valid certs!
+            if (ApnsUtil.checkValidity(iOSVariant.getCertificate(), iOSVariant.getPassphrase().toCharArray())) {
```

Comment:
since this is done in `buildApnsClient` it should only be checked once per cached client right? (So no check for every batch, which would be unneccessary)